pymc3/tests/test_step.py

```diff
@@ -456,3 +456,16 @@ def test_linalg(self, caplog):
             error.match('issues during sampling')
 
             assert not trace.report.ok
+
+    def test_sampler_stats(self):
+        with Model():
+            x = Normal('x', mu=0, sd=1)
+            trace = sample(draws=10, tune=1, chains=1)
+
+        expected_stat_names = {'depth', 'diverging', 'energy', 'energy_error',
+            'logp', 'max_energy_error', 'mean_tree_accept', 'step_size',
+            'step_size_bar', 'tree_size', 'tune'}
+
```

Comment:
Should also add test to check if the saved logp is the same compare to computing the logp post sample via passing the point from traces to logp_func

Reply:
I see, where exactly is `logp_func` and a list of the model's states? I can't find them in `model`...

Reply:
What i mean is to compare the saved logp with:
```python
logp = model.logp
logp_ = np.array([logp(trace.point(i,chain=c)) for c in trace.chains for i in range(len(trace))]) 
```
as they should be the same. The same test could also be use when we implemented recording of logp for `CompoundStep`.

Reply:
Added!